controllers/nicclusterpolicy_controller.go

```diff
@@ -141,58 +144,68 @@ func (r *NicClusterPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Req
+// returns true if requeue (resync) is required
 func (r *NicClusterPolicyReconciler) handleMOFEDWaitLabels(
-	ctx context.Context, cr *mellanoxv1alpha1.NicClusterPolicy) error {
-	if cr.Spec.OFEDDriver != nil {
-		pods := &corev1.PodList{}
-		podLabel := "mofed-" + cr.Spec.OFEDDriver.Version
-		_ = r.Client.List(ctx, pods, client.MatchingLabels{"driver-pod": podLabel})
-		for i := range pods.Items {
-			pod := pods.Items[i]
-			labelValue := "true"
-			// We assume that OFED pod contains only one container to simplify the logic.
-			// We can revisit this logic in the future if needed
-			if len(pod.Status.ContainerStatuses) != 0 && pod.Status.ContainerStatuses[0].Ready {
-				labelValue = "false"
-			}
-			if err := setOFEDWaitLabel(ctx, r.Client, pod.Spec.NodeName, labelValue); err != nil {
-				return err
-			}
-		}
-	} else {
+	ctx context.Context, cr *mellanoxv1alpha1.NicClusterPolicy) (bool, error) {
+	reqLogger := log.FromContext(ctx)
+	if cr.Spec.OFEDDriver == nil {
+		reqLogger.V(consts.LogLevelDebug).Info("no OFED config in the policy, check OFED wait label on nodes")
 		return r.handleMOFEDWaitLabelsNoConfig(ctx)
 	}
-
-	return nil
+	pods := &corev1.PodList{}
+	podLabel := "mofed-" + cr.Spec.OFEDDriver.Version
+	_ = r.Client.List(ctx, pods, client.MatchingLabels{"driver-pod": podLabel})
+	for i := range pods.Items {
+		pod := pods.Items[i]
+		labelValue := "true"
+		// We assume that OFED pod contains only one container to simplify the logic.
+		// We can revisit this logic in the future if needed
+		if len(pod.Status.ContainerStatuses) != 0 && pod.Status.ContainerStatuses[0].Ready {
+			reqLogger.V(consts.LogLevelDebug).Info("OFED Pod is ready on the node",
+				"node", pod.Spec.NodeName)
+			labelValue = "false"
+		}
+		if err := setOFEDWaitLabel(ctx, r.Client, pod.Spec.NodeName, labelValue); err != nil {
+			return true, err
+		}
+	}
+	return false, nil
 }
 
 // handleMOFEDWaitLabelsNoConfig handles MOFED wait label for scenarios when OFED is
 // not configured in NicClusterPolicy, do the following:
 // - set "network.nvidia.com/operator.mofed.wait" to false on Nodes with NVIDIA NICs
 // - remove "network.nvidia.com/operator.mofed.wait" which have no NVIDIA NICs anymore
 // - set "network.nvidia.com/operator.mofed.wait" to true if detects OFED Pod
-// on the node (probably in the terminating state)
-func (r *NicClusterPolicyReconciler) handleMOFEDWaitLabelsNoConfig(ctx context.Context) error {
+// on the node (probably in the terminating state).
+// returns true if requeue (resync) is required
+func (r *NicClusterPolicyReconciler) handleMOFEDWaitLabelsNoConfig(ctx context.Context) (bool, error) {
+	reqLogger := log.FromContext(ctx)
 	nodesWithOFEDContainer := map[string]struct{}{}
 	pods := &corev1.PodList{}
 	if err := r.Client.List(ctx, pods, client.MatchingLabels{consts.OfedDriverLabel: ""}); err != nil {
-		return errors.Wrap(err, "failed to list pods")
+		return true, errors.Wrap(err, "failed to list pods")
```

Comment:
in case of error first return param should be ignored by the caller, maybe just return false in case of error as its the default for bool ?

Reply:
some thought drill: what would you return in case the resync val was *bool ? nil ?

Reply:
Agree. I update the code to return false in case of error